src/libostree/ostree-repo.c

```diff
@@ -3292,6 +3292,28 @@ ostree_repo_get_mode (OstreeRepo  *self)
   return self->mode;
 }
 
+/**
+ * ostree_repo_get_min_free_space:
+ * @self: Repo
+ * @error: error
+ *
+ * Before this function can be used, ostree_repo_prepare_transaction() must have been
+ * called.
```

Comment:
Oh yeah, How did I miss this. Thank you.

Thinking on this, we want to only read the value of `min-free-space-*` only when there is an ongoing transaction (`repo->in_transaction`) because that's where the config option is actually computed. If we read outside (e.g. after `ostree_repo_commit_transaction`) it could be inconsistent as the config option might get changed and one will still be reading the old value.